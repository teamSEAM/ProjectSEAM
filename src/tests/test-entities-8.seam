# Requires WHILE to work
Lots of things going on
Spawn Entity (brings in things into the step function so will be printing)
Decrement function that will change the x that is in a higher scope
Return x after decrementing
#
entity Ent_1:
	int x
	func start():
		screen.out("Hi I am 1!!")
		x=5
	func step():
		screen.out("One")
	func decrement():
		x=x-1
entity World:y
	int count
	<Ent_1> one
	func start():
		one = spawn Ent_1
	func step():
		count=(one.decrement()).x
		if(count<0):
			kill one

