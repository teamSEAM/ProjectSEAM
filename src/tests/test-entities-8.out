One
One
One
One
One
